src/atoms/TextInput/TextInput.native.js

```diff
@@ -249,6 +249,10 @@ const TextInput = ({
     [layoutDimensions, setLayoutDimensions],
   );
 
+  useEffect(() => {
+    if (!isEmpty(children)) setInput(children);
```

Comment:
What is the use-case?

Reply:
This makes the component controlled. So when you pass children to text input, it will use that as the value to render but also give you the flexibility to edit it and send updates via ```onChangeText```

Our use case here is to render OTP text field with empty state and update it after retrieving from sms

Reply:
Got it.